common/djangoapps/third_party_auth/views.py

```diff
@@ -40,12 +40,15 @@ def saml_metadata_view(request):
     Get the Service Provider metadata for this edx-platform instance.
     You must send this XML to any Shibboleth Identity Provider that you wish to use.
     """
+    backend_name = request.GET.get('backend_name', 'tpa-saml')
+    if backend_name not in _PSA_SAML_BACKENDS:
+        raise Http404
     if not SAMLConfiguration.is_enabled(request.site):
         raise Http404
-    complete_url = reverse('social:complete', args=("tpa-saml", ))
+    complete_url = reverse('social:complete', args=(backend_name, ))
```

Comment:
Argh, I didn't anticipate that we'd need different metadata for different backends - definitely a downside of this approach. I think your solution here is reasonable though.

In the future, we could improve this situation by adding a new URL like `/auth/complete/saml/` that auto-detects the backend and then passes the request through to the `social:complete` view, so that separate metadata is no longer required.